esp/public/media/scripts/onsite/teacher_checkin.js

```diff
@@ -123,16 +123,30 @@ $j(function(){
         });
     }
 
-    $j(".text").click(function(){
-        var username = $j(this).data("username");
-        var section = $j(this).data("section");
-        var $td = $j(this.parentNode);
-        var $msg = $td.children('.message');
-        textTeacher(username, section, function(response) {
-            $msg.text(response.message);
+    function textTeacher(btn) {
+        var username = $j(btn).data("username");
+        var section = $j(btn).data("section");
+        var msg = $j(btn).closest('td').find('.message');
+        sendText(username, section, function(response) {
+            $j(msg).text(response.message);
         });
-        $j(this).attr("disabled",true);
-        $msg.text('Texting teacher...');
+        $j(btn).attr("disabled",true);
+        $j(msg).text('Texting teacher...');
+    }
+
+    $j(".text").click(function(){
+        textTeacher($j(this))
+    });
+
+    $j(".text-all").click(function(){
+        var r = confirm("Are you sure you'd like to text ALL unchecked-in teachers?");
```

Comment:
Here you could even mention the number of teachers, so someone doesn't accidentally click it on the "all teachers" page if they didn't intend.